cmake/toolchain/arcmwdt/generic.cmake

```diff
@@ -23,7 +23,17 @@ find_package(Zephyr-sdk 0.15 REQUIRED)
 # https://github.com/zephyrproject-rtos/sdk-ng/pull/682 got merged and we switch to proper SDK
 # version.
 set(TOOLCHAIN_HOME ${ZEPHYR_SDK_INSTALL_DIR})
-include(${ZEPHYR_SDK_INSTALL_DIR}/cmake/zephyr/target.cmake)
+
+# On very early stage build system needs to get access to DTC preprocessor.
+# MWDT has no it's own preprocessor, so here zephyr-SDK preprocessor is used.
+# At the same time zephyr-SDK requires ARCH variable to be initialized before include "generic.cmake",
+# but in hew HWMv2 model ARCH variable will be initialized more later.
+# This workaround uses any (first awailable, independent on ARCH) toolchain from SDK for DTC preprocessing only.
+# For other build stages ARCMWDT will be used.
+include(${ZEPHYR_SDK_INSTALL_DIR}/cmake/zephyr/generic.cmake)
+unset(TOOLCHAIN_HAS_NEWLIB CACHE)
+unset(TOOLCHAIN_HAS_PICOLIBC CACHE)
```

Comment:
Have added this to #71368.
Approving the use of generic.cmake over target.cmake, which should actually have been the case from the start.